Automattic/Automattic-Tracks-iOS
# Adds Reachability for resilience & WiFi detection

Closes #2 

Reachability has been added to help with two specific tasks:
1. Detect when WiFi is in use for the `device_info_wifi_connected` property. Up until now every event has been set to NO.
2. Disable the timer to send events when the network is detected as being offline. This helps reduce potential battery drain continually attempting to hit the host.

**Testing:**
Use the TracksDemo application. The easiest way to test this is probably by adding an `NSLog` statement to `StatsServiceRemote` to dump out the `dataToSend` contents. You can then verify the right flags are attached to the events being sent to Tracks. Possible scenarios:

1. Turn on WiFi, record an event.
2. Turn off WiFi (wait like 2 secs for Reachability to detect) and record an event.
3. Turn off cellular + WiFi, record an event.
4. Turn off all data (airplane mode), record an event.

Depending on the device you're testing and if you have cellular and WiFi or just WiFi these scenarios will be a little different. Here is the expected outcome:

1. No data at all - Events will get recorded but WiFi will be listed as NO and the remote call & timer should never fire.
2. Only cellular - Events will get recorded, WiFi is listed as NO. Remote call should still happen, timer remains on and firing every 10sec.
3. Only WiFi - Events will get recorded, WiFi is listed as YES. Remote call should still happen, timer remains on and firing every 10sec.
4. Both cellular + WiFi - Events will get recorded, WiFi is listed as YES. Remote call should still happen, timer remains on and firing every 10sec.
5. App background - events should stop with one last attempt immediately upon backgrounding.

Needs Review: @jleandroperez 
cc: @daniloercoli
